### DISCUSSION TITLE
getting  blank input value on keydown event in fluent ui seachbox

### DISCUSSION THREAD

**Discussion by sagarkalokhe09** [microsoft/fluentui#28427] [Q&A] [answered]
Hi Team

I am trying to get input value from fluent ui searchbox using onkeydown event. but some how i am always getting an blank value.

I am simply using the onkeyDown prop from the searchbox like this

<SearchBox onKeyDown={(e)=>{alert(e.currentTarget.value)}}>

**Comment by behowell** [comment, marked as answer]
Hi @sagarkalokhe09, is this the v8 SearchBox? https://developer.microsoft.com/en-us/fluentui#/controls/web/searchbox

The `onKeyDown` event happens before the value is updated. Consider using the `onChange` event instead. That also has a second parameter, which is the new value. See the documentation on the link above.

```jsx
<SearchBox onChange={(_ev, value) => console.log(value)} />
```

**Reply by ZzyzxIsAlreadyTaken** [reply]
Still no value.
